src/renderer/page/publish/index.js

```diff
@@ -4,40 +4,82 @@ import { doNavigate, doHistoryBack } from 'redux/actions/navigation';
 import { doClaimRewardType } from 'redux/actions/rewards';
 import {
   selectMyClaims,
-  selectFetchingMyChannels,
-  selectMyChannelClaims,
   selectClaimsByUri,
+  selectClaimById
 } from 'redux/selectors/claims';
 import { selectResolvingUris } from 'redux/selectors/content';
+import { selectPublishFormValues } from 'redux/selectors/publish';
 import {
-  doFetchClaimListMine,
-  doFetchChannelListMine,
   doResolveUri,
-  doCreateChannel,
-  doPublish,
 } from 'redux/actions/content';
 import { selectBalance } from 'redux/selectors/wallet';
-import rewards from 'rewards';
+import { makeSelectFileInfoForUri } from 'redux/selectors/file_info';
+import {
+  doUpdateFilePath,
+  doClearFilePath,
+  doUpdateContentPricePref,
+  doUpdateContentPrice,
+  doUpdateContentDescriptors,
+  doClearPublish,
+  doToggleTos,
+  doUpdatePublishForm,
+  doPublish,
+  doClearPublishError
+} from 'redux/actions/publish';
 import PublishPage from './view';
 
-const select = state => ({
-  balance: selectBalance(state),
-  myClaims: selectMyClaims(state),
-  fetchingChannels: selectFetchingMyChannels(state),
-  channels: selectMyChannelClaims(state),
-  claimsByUri: selectClaimsByUri(state),
-  resolvingUris: selectResolvingUris(state),
-});
+const select = (state, props) => {
+  const publishState = state.publish;
+  const { uri, name } = publishState;
+  const { params: { id } } = props;
+
+  let prefillClaim;
+  if (id) {
+    prefillClaim = selectClaimById(id)(state);
```

Comment:
I think I'm probably going to change this and have whoever links to the publish page be responsible for populating the form before navigating. Currently there can be issues with this if you navigate away from the form and come back to an edit.

possibly
```
// file page
populatePublishForm(fileInfo);
doNavigate('/publish')
```

Reply:
:+1: